test/unittests/bitcount_test_cases.hpp

```diff
@@ -9,7 +9,7 @@
 
 namespace fizzy::test
 {
-constexpr std::pair<uint32_t, uint32_t> popcount32_test_cases[]{
+constexpr std::pair<uint32_t, int> popcount32_test_cases[]{
```

Comment:
Duplicating inputs does not make sense. Create a struct with results for all instructions.
```
template <T>
struct SomeGoodName
{
   T input;
   int popcount;
   int countl_zero;
   int countr_zero;
};

constexpr SomeGoodName<uint32_t> bit_counting32_test_cases[]{
    {0, 0, 32, 32},
    {0x80, 1, 24, 7},
    ...
};

Reply:
I will keep this a separate commit even for merging, as not sure it is more readable alltogether.